lib/SIL/IR/SILFunction.cpp

```diff
@@ -920,16 +916,72 @@ bool SILFunction::hasName(const char *Name) const {
   return getName() == Name;
 }
 
+/*
+ Checks if this (callee) function body can be inlined into the caller
+ by comparing their SerializedKind_t values. 
+ 
+ If the \p assumeFragileCaller is true, the caller must be serialized,
+ in which case the callee needs to be serialized also to be inlined.
+ If both callee and caller are not_serialized, the callee can be inlined
+ into the caller during SIL inlining passes even if it (and the caller)
+ might contain private symbols. If this callee is serialized_for_pkg, it
+ can only be referenced by a serialized caller but not inlined into it.
+ 
+ canInlineInto:                                 Caller
+                            | not_serialized | serialized_for_pkg | serialized
+        not_serialized      |      ok        |       no           |    no
+Callee  serialized_for_pkg  |      ok        |       ok           |    no
+        serialized          |      ok        |       ok           |    ok
+
+*/
+bool SILFunction::canBeInlinedIntoCaller(
+      std::optional<SerializedKind_t> callerSerializedKind,
+      bool assumeFragileCaller) const {
```

Comment:
I don't understand the need of the boolean `assumeFragileCaller`. 

If the caller is fragile (i.e serialized) this should be communicated via passing `SerializedKind_t::Serialized`

That argument should not be optional.